File: aab-web-application/_js/networking/webSocket/clientConnectionType.js

/**
 * Defines the possible client connection types.
 * @type {{NotDefined: string, MobileDevice: string, Server: string, Desktop: string, EmulationDevice: string, HoloLens: string, Wearable: string}}
 */
const ClientConnectionType = {
    "NotDefined": "NotDefined",
    "Server": "Server",
    "EmulationDevice": "EmulationDevice",

    "Wearable": "Wearable",
    "HoloLens": "HoloLens",
    "MobileDevice": "MobileDevice",
    "Desktop": "Desktop"
}

export {
    ClientConnectionType
}

File: aab-hl-application/Assets/00 BasicCode/1 Script/Networking/WebSocket/ClientConnectionType.cs
namespace MasterNetworking.WebSocket
{
    public enum ClientConnectionType
    {
        NotDefined,
        Server,
        EmulationDevice,

        Wearable,
        HoloLens,
        MobileDevice,
        Desktop
    }

    public static class ClientConnectionTypeUtils
    {
        public static string ToString(ClientConnectionType ccType)
        {
            switch (ccType)
            {
                case ClientConnectionType.Server:            return "Server";
                case ClientConnectionType.EmulationDevice:   return "EmulationDevice";
                case ClientConnectionType.Wearable:          return "Wearable";
                case ClientConnectionType.HoloLens:          return "HoloLens";
                case ClientConnectionType.MobileDevice:      return "MobileDevice";
                case ClientConnectionType.Desktop:           return "Desktop";
            }

            return null;
        }

        public static ClientConnectionType? FromString(string ccType)
        {
            switch (ccType)
            {
                case "Server":          return ClientConnectionType.Server;
                case "EmulationDevice": return ClientConnectionType.EmulationDevice;
                case "Wearable":        return ClientConnectionType.Wearable;
                case "HoloLens":        return ClientConnectionType.HoloLens;
                case "MobileDevice":    return ClientConnectionType.MobileDevice;
                case "Desktop":         return ClientConnectionType.Desktop;
            }

            return null;
        }
    }
}
